### ISSUE TITLE
Add branch/suffix flag to support resource file versioning

### ISSUE THREAD

**Issue by sconno05** [transifex/transifex-client#116]
Consider the scenario when a developer is working on a git branch and they need to add/modify/remove keys on a source file (e.g. web_source.json).

At the moment it is dangerous to run `tx push -s` from the branch because that could prematurely remove keys that are still being used by the master branch (and production).  Instead, it would be nice to add a flag to the push command that creates new "branched" resource files in Transifex:

`tx push -s --resource="test-website.web_sourcejson" --branch="branch1"`

The above command would result in a new resource file called `web_source_branch1.json` in Transifex, while the original/master/production resource is unchanged.  That resource file can then be translated and tracked to 100% completion separately.  Once complete, a developer can run:

`tx pull -a --resource="test-website.web_sourcejson" --branch="branch1"`

...and if you are so inclined, you can wrap this command with a custom alias (e.g. `rake tx:push-branch` and `rake tx:pull-branch`) that automatically plugs in the current git branch.

I understand "branch" may not be the proper nomenclature.  Other thoughts are "version", "tag", "suffix".

**Comment by kouk** [comment]
Thank you for the suggestion @sconno05! I think it's a great idea to make the client do some things more automatically, similar to what you propose. I will add this to our TODO list. In the meantime you can achieve a similar thing by following these steps:

1. edit `./.tx/config` and copy the section that describes your resource. E.g. in my test project I duplicated the resource section and ended up with:
   
   ```
   [main]
   host = https://www.transifex.com

[kk3-web.newresource]
file_filter = path/to/<lang>/LC_MESSAGES/myproject.po
source_file = path/to/en/LC_MESSAGES/myproject.po
source_lang = en
type = PO

[kk3-web.newresource_branch1]
file_filter = path/to/<lang>/LC_MESSAGES/myproject.po
source_file = path/to/en/LC_MESSAGES/myproject.po
source_lang = en
type = PO
   ```
2. Then do:
    
    ```
 $ tx push -s -r "kk3-web.newresource_branch1"
Pushing translations for resource kk3-web.newresource_branch1:
Pushing source file (transifex/locale/en/LC_MESSAGES/django.po)
Resource does not exist.  Creating...
Done.
    ```

**Comment by sconno05** [comment]
Yes, thank you.  I was planning on using a workaround just like this.  I did raise a concern with your sales rep that automating this for every branch could increase our key count (and therefore cost) substantially.  He recommended just pushing a diff to Transifex, which I think is a great idea.  I'm not sure if this functionality needs to be part of the transifex-client.  I think we could manage that ourselves in an automated way (download the source from Transifex, run a diff between that and whats on the branch and then push the diff).

I'm not sure yet how we will generate the diff, but I don't think it will be that hard to figure out.

**Comment by kouk** [comment]
@sconno05 thanks for the great feedback
